src/Service/Services/OpenAPI/OpenApiDocumentor.cs

```diff
@@ -0,0 +1,708 @@
+        /// <returns>Primary Key path component. Empty string if no primary keys exist on database object source definition.</returns>
+        private Tuple<string, List<OpenApiParameter>> CreatePrimaryKeyPathComponentAndParameters(string entityName)
+        {
+            SourceDefinition sourceDefinition = _metadataProvider.GetSourceDefinition(entityName);
+            List<OpenApiParameter> parameters = new();
+            StringBuilder pkComponents = new();
+
+            // Each primary key must be represented in the path component.
+            foreach (string column in sourceDefinition.PrimaryKey)
+            {
+                string columnNameForComponent = column;
+
+                if (_metadataProvider.TryGetExposedColumnName(entityName, column, out string? mappedColumnAlias) && !string.IsNullOrEmpty(mappedColumnAlias))
+                {
+                    columnNameForComponent = mappedColumnAlias;
+                }
+
+                if (sourceDefinition.Columns.TryGetValue(columnNameForComponent, out ColumnDefinition? columnDef))
+                {
+                    OpenApiSchema parameterSchema = new()
+                    {
+                        Type = columnDef is not null ? SystemTypeToJsonDataType(columnDef.SystemType) : string.Empty
+                    };
+
+                    OpenApiParameter openApiParameter = new()
+                    {
+                        Required = true,
+                        In = ParameterLocation.Path,
+                        Name = $"{columnNameForComponent}",
+                        Schema = parameterSchema
+                    };
+
+                    parameters.Add(openApiParameter);
+                    string pkComponent = $"/{columnNameForComponent}/{{{columnNameForComponent}}}";
+                    pkComponents.Append(pkComponent);
+                }
+            }
+
+            return new(pkComponents.ToString(), parameters);
+        }
+
+        /// <summary>
+        /// Determines whether the database object has an autogenerated primary key
+        /// used to distinguish which request and response objects should reference the primary key.
+        /// </summary>
+        /// <param name="sourceDefinition">Database object metadata.</param>
+        /// <returns>True, when the primary key is autogenerated. Otherwise, false.</returns>
+        private static bool SourceContainsAutogeneratedPrimaryKey(SourceDefinition sourceDefinition)
+        {
+            bool sourceObjectHasAutogeneratedPK = false;
+            // Create primary key path component.
+            foreach (string column in sourceDefinition.PrimaryKey)
+            {
+                string columnNameForComponent = column;
+
+                if (sourceDefinition.Columns.TryGetValue(columnNameForComponent, out ColumnDefinition? columnDef) && columnDef is not null && columnDef.IsAutoGenerated)
+                {
+                    sourceObjectHasAutogeneratedPK = true;
```

Comment:
break; after finding at least 1 PK thats autogen?

Reply:
yep, also clarifying the comment with an example and rewording to be less vague.

Reply:
Lets say a PK has some cols that are autogen and some that are not. In this case, would the PUT/PATCH/POST be required to specify the PK columns that are not autogenerated?

Reply:
updated to create a component schema definition with no autogenerated PKs and still include non-autogenerated PK's so that POST requests can define non-autogenerated PK's in the request body.